### PR TITLE
New Email for Tribe Members about Hackathon Prizes

### LINKED ISSUE #7140: Email tribe members about hackathon prizes
### User Story

As a tribe member/follower, I'd like to know when an organization iii participating in a Hackathon 

### Why Is this Needed

no email notification exists

If we build this email, developers will be notified and sign up, then sponsors will have more registrants and hackers working on their hackathon prizes.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: feature

### Definition of Done
Email Details

### User Story

[comment]: # (As a <user type>, I want to <task> so that <goal>.)

As a tribe member, I want to know the prizes from the tribe when participating in a hackathon

As a sponsor, I want my tribe members to automatically be notified the prizes available when I'm sponsoring a hackathon so that they are interested ot participate

### Why Is this Needed

[comment]: # (Describe the problem and why this task is needed. Provide description of the current state, what you would like to happen, and what actually happen)
*Summary*:

IF we engage with developers already interested an org, then sponsors will have a great hackathon turnout, because developers will participate in the sponsored hackathon.

IF we do this, it'll Increase number of hackers registering for a hackathon.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

If a sponsor signs up for a hackathon, email to current tribes members and let them know that the prizes offered by X sponsor in a hackathon

### Current Behavior
[comment]: # (Describe what actually happened.)

No automatic email

### Expected Behavior
[comment]: # (Describe what you expected to happen.)
Automatic email sends out as soon as a hackathon starts

### Definition of Done
[comment]: # (Any other information that would be useful, bullets are helpful.)

- [ ] Build email below
- [ ] ensure email  is responsive (see other emails for example)
- [ ] the image displayed "clarity hackthon"  will be dynamic based on the hackathon the sponsor  is part of.
       note the image should leverage the same image from the hackathon onboarding page.
- [ ] join button should take users to  the  specific  hackathon's  onboarding /landing page.
- [ ] this email should trigger  for  all  hackathon sponsors 3  days prior to hackathon starting
- [ ] view prize should  go  to specific prize page
- [ ] add ability to fire this from admin

### Data Requirements
Increase in number of registered hackers per hackathon

### Additional Information

85852050-73b5af00-b7e2-11ea-8992-c38b7f98ebf0

### PR DESCRIPTION
**Pull request by sebastiantf**
##### Description
This PR builds a new email that will be sent to the tribe members of a hackathon sponsor 3 days prior to the hackathon start date, listing out the prizes by that sponsor

##### Refers/Fixes
Fixes #7140 

##### Screenshots (Updated after rewrite)
**Gmail Web:**

tribe_hackathon_prizes_gmail_web_rewrite

**Gmail Mobile:**

tribe_hackathon_prizes_gmail_mobile_rewrite

**Web Preview:**

tribe_hackathon_prizes_web_rewrite

### CODE REVIEW

**Comment by danlipert** [comment]
@PixelantDesign @connoroday This will send multiple emails to the same user if they are members of multiple tribes that are participating in the hackathon - we need to combine the emails so each user only gets one I think @sebastiantf

**Comment by sebastiantf** [comment]
Combining all the bounties by all the sponsors/tribes the user is part of, might make the email a lot big in content. Aren't most users part of a lot number of tribes? Just my thoughts. What do you think @PixelantDesign?

**Comment by PixelantDesign** [comment]
If the user is part of two tribes participating in the same hackathon, they should receive just 1 email if possible.

**Comment by sebastiantf** [comment]
> If the user is part of two tribes participating in the same hackathon, they should receive just 1 email if possible.

If that's the case, I would have to rewrite what I've did now for this email. And I would need some time for that

**Comment by PixelantDesign** [comment]
@sebastiantf are you still interested in this ticket?

**Comment by sebastiantf** [comment]
Yeah @PixelantDesign. I'm a bit caught up is all. Will get on this asap

**Comment by sebastiantf** [comment]
@PixelantDesign @danlipert Extremely sorry for the delay. I have rewritten the logic to combine all bounties of all tribes of a member into single email. Please review

**Comment by PixelantDesign** [comment]
Thanks @sebastiantf Looks good!

Just to clarify -
When a hackathon starts users who **follow the organization** will receive this email that explains the prizes associated with the hackathon, correct?

Not all users will receive this email.

**Comment by sebastiantf** [comment]
Yes @PixelantDesign. Only the tribe members of the org who are sponsors in the hackathon would receive this email

**Comment by PixelantDesign** [comment]
great! are you able to provide a video or screenshot of this?

**Comment by sebastiantf** [comment]
> great! are you able to provide a video or screenshot of this?

I have already updated the screenshots in the opening comment: https://github.com/gitcoinco/web/pull/7146#issue-456440294

**Review by connoroday — APPROVED** [review]

**Comment by frankchen07** [comment]
LGTM as long as we don't have duplicate emails across multiple tribes.

Is there an option for the user to opt-out of tribe specific communications or is that out of scope for this ticket?

**Review by frankchen07 — COMMENTED** [review]
LGTM as long as we don't have duplicate emails across multiple tribes, which sounds like it was fixed.

Last item: is there an option for the user to opt-out of tribe specific communications or is that out of scope for this ticket?

**Comment by PixelantDesign** [comment]
@frankchen07 I think I'll open another ticket for that....@sebastiantf what do you think? A simple checkbox in the email tab of the settings should do it.

**Comment by sebastiantf** [comment]
> @frankchen07 I think I'll open another ticket for that....@sebastiantf what do you think? A simple checkbox in the email tab of the settings should do it.

Yeah @PixelantDesign guess a simple setting to keep along with the other email settings

**Comment by willsputra** [comment]
looks good to me!

**Review by willsputra — APPROVED** [review]

**Comment by PixelantDesign** [comment]
@thelostone-mc could we have a review on this? it's for October

**Comment by thelostone-mc** [comment]
@sebastiantf gimme a ping when you resolve the conflicts and we can get this in

**Comment by sebastiantf** [comment]
@thelostone-mc I have resolved the conflicts and pushed again

File: app/marketing/mails.py [hunk 1]
```diff
@@ -1997,3 +2011,47 @@ def remember_your_cart(profile, cart_query, grants, hours):
             send_mail(from_email, to_email, subject, text, html, categories=['marketing', func_name()])
     finally:
         translation.activate(cur_language)
+
+def tribe_hackathon_prizes(hackathon):
+    from dashboard.models import TribeMember, Sponsor
+    from marketing.utils import generate_hackathon_email_intro
+
+    sponsors = hackathon.sponsors.all()
```
Comment (octavioamu) [inline app/marketing/mails.py:2019]: we have a problem here, should use sponsor_profiles we not longer use Sponsors to control hackathons

**Review by octavioamu — COMMENTED** [review]
doing regression test I saw this problem, can you submit a PR to fix this using sponsor_profiles?

**Comment by PixelantDesign** [comment]
@sebastiantf can you take a look please? :)

File: app/marketing/mails.py (hunk 1 above)
Comment (sebastiantf) [reply app/marketing/mails.py:2019]: @octavioamu @PixelantDesign I have a PR up at #7589

**Pull request merged**